browser.js

```diff
@@ -15,9 +16,11 @@ exports.pbkdf2 = function (password, salt, iterations, keylen, digest, callback)
   })
 }
 
+var defaultEncoding = (process.browser || pVersionMajor >= 6) ? 'utf-8' : 'binary'
+
 exports.pbkdf2Sync = function (password, salt, iterations, keylen, digest) {
-  if (!Buffer.isBuffer(password)) password = new Buffer(password, 'binary')
-  if (!Buffer.isBuffer(salt)) salt = new Buffer(salt, 'binary')
+  if (!Buffer.isBuffer(password)) password = new Buffer(password, defaultEncoding)
+  if (!Buffer.isBuffer(salt)) salt = new Buffer(salt, defaultEncoding)
```

Comment:
Just the situation,  wondering if you had any thoughts on impl.
Otherwise,  I assume this PR is good to go now